modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/buttons/RowButton.kt

```diff
@@ -48,6 +51,88 @@ fun RowButton(
     cornerRadius: Dp = 12.dp,
     textVerticalPadding: Dp = 6.dp,
     @DrawableRes textIcon: Int? = null,
+) {
+    BaseRowButton(
+        text = text,
+        modifier = modifier,
+        includePadding = includePadding,
+        enabled = enabled,
+        border = border,
+        colors = colors,
+        textColor = textColor,
+        fontFamily = fontFamily,
+        fontWeight = fontWeight,
+        leadingIcon = if (leadingIcon != null) {
+            {
+                Icon(
+                    painter = painterResource(leadingIcon),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .padding(4.dp),
+                    tint = textColor,
+                )
+            }
+        } else null,
+        onClick = onClick,
+        cornerRadius = cornerRadius,
+        textVerticalPadding = textVerticalPadding,
+        textIcon = textIcon,
+    )
+}
+
+@Composable
+fun RowLoadingButton(
+    text: String,
+    modifier: Modifier = Modifier,
+    includePadding: Boolean = true,
+    enabled: Boolean = true,
+    border: BorderStroke? = null,
+    isLoading: Boolean = false,
+    colors: ButtonColors = ButtonDefaults.buttonColors(),
+    textColor: Color = MaterialTheme.theme.colors.primaryInteractive02,
+    onClick: () -> Unit
+) {
+    BaseRowButton(
+        text = text,
+        modifier = modifier,
+        includePadding = includePadding,
+        enabled = enabled,
+        border = border,
+        colors = colors,
+        textColor = textColor,
+        leadingIcon = if (isLoading) {
+            {
+                CircularProgressIndicator(
+                    modifier = Modifier
+                        .size(24.dp),
+                    color = textColor,
+                    strokeWidth = 2.dp
+                )
+            }
+        } else null,
+        onClick = onClick@{
+            if (isLoading) return@onClick
+            onClick()
+        },
+    )
+}
+
+@Composable
+fun BaseRowButton(
```

Comment:
Nitpick: Can we make it private?

Reply:
Kept it public since I have moved the three buttons to their own files